base/sparse/sparsematrix.jl

```diff
@@ -942,8 +943,9 @@ function getindex{Tv,Ti<:Integer}(A::SparseMatrixCSC{Tv,Ti}, I::Range, J::Abstra
     # Form the structure of the result and compute space
     for j = 1:nJ
         @inbounds col = J[j]
+        1 <= col <= n || throw(BoundsError())
```

Comment:
This may not be required as `colptrA[co+1]` in the for loop condition would have done the check. Alternatively, we can probably have this check and mark the outer for loop `@inbounds`?